projects/netgrif-components/schematics/view/create-view-prompt/views/public-resolver-view/create-public-resolver-view.ts

```diff
@@ -0,0 +1,45 @@
+import {chain, Rule, Tree} from '@angular-devkit/schematics';
+import {strings} from '@angular-devkit/core';
+import {createFilesFromTemplates, createRelativePath, getProjectInfo} from '../../../../_utility/utility-functions';
+import {updateAppModule} from '../../../_utility/view-utility-functions';
+import {addViewToViewService} from '../../../_utility/view-service-functions';
+import {TabbedView} from '../../models/tabbed-view';
+import {ViewClassInfo} from '../../../../_commons/view-class-info';
+import {ImportToAdd} from '../../../../_commons/import-to-add';
+import {CreateTaskViewArguments} from '../../models/create-task-view-arguments';
+
+
+export function createPublicResolverView(tree: Tree, args: CreateTaskViewArguments & TabbedView, addViewToService: boolean): Rule {
+    const projectInfo = getProjectInfo(tree);
+    const view = new ViewClassInfo(
+        args.path,
+        args.viewType,
+        args.componentName);
+    const rules = [];
+
+    const templateParams = {
+        prefix: projectInfo.projectPrefixDasherized,
+        className: view.nameWithoutComponent,
+        viewPath: args.path,
+        dasherize: strings.dasherize,
+        classify: strings.classify,
+        configName: projectInfo.projectNameClassified,
+        configImportPath: createRelativePath(view.fileImportPath, `./${projectInfo.projectNameDasherized}-configuration.service`),
+        isDefaultTabbedTaskView: !!args.isDefaultTabbedTaskView
+    };
+
+    const commonPathPrefix = './views/public-resolver-view/files/';
+    const destinationPath = `${projectInfo.path}/views/${args.path}`;
+    rules.push(createFilesFromTemplates(`${commonPathPrefix}common`, destinationPath, templateParams));
+
+    updateAppModule(tree, view.className, view.fileImportPath, [
+        new ImportToAdd('FlexModule', '@angular/flex-layout'),
+        new ImportToAdd('FlexLayoutModule', '@angular/flex-layout'),
+        new ImportToAdd('MaterialModule', '@netgrif/components-core')
```

Comment:
`@angular/router` is used in the generated template files. The corresponding module import should be added to the project

Reply:
in our guide to install nae and setup the project is written, that you should create a project with router, sou you get the import there. Anyway, there is a starter project that already had nae installed and all imports imported